components-core/src/main/java/com/adyen/checkout/components/core/CheckoutConfiguration.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 5/12/2023.
+ */
+
+package com.adyen.checkout.components.core
+
+import android.annotation.SuppressLint
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.CONTENTS_FILE_DESCRIPTOR
+import androidx.annotation.RestrictTo
+import com.adyen.checkout.components.core.internal.Configuration
+import com.adyen.checkout.core.Environment
+import kotlinx.parcelize.IgnoredOnParcel
+import java.util.Locale
+
+class CheckoutConfiguration(
+    override val shopperLocale: Locale,
+    override val environment: Environment,
+    override val clientKey: String,
+    override val amount: Amount? = null,
+    override val analyticsConfiguration: AnalyticsConfiguration? = null,
+    @IgnoredOnParcel
+    private val config: CheckoutConfiguration.() -> Unit = {},
```

Comment:
Can we use a better name for this?

Reply:
Changed it to `configurationBlock`. Is that okay?